frame/session/src/lib.rs

```diff
@@ -156,6 +156,8 @@ pub trait SessionManager<ValidatorId> {
 	/// `new_index` is strictly greater than from previous call.
 	///
 	/// The first session start at index 0.
+	///
+	/// `new_session(session)` is guaranteed to be called before `end_session(session-1)`.
```

Comment:
this is not related to the bug, but as long as staking implement SessionManager with this constraint, I think it must stay here.